src/ORM/DatabaseAdmin.php

```diff
@@ -390,6 +373,84 @@ public function doBuild($quiet = false, $populate = true, $testMode = false)
         ClassInfo::reset_db_cache();
     }
 
+    /**
+     * Given a base data class, a field name and an old and new class name (value), look for obsolete ($oldClassName)
+     * values in the $dataClass's $fieldName column and replace it with $newClassName.
+     *
+     * @param string $dataClass    The data class to look up
+     * @param string $fieldName    The field name to look in for obsolete class names
+     * @param string $oldClassName The old class name
+     * @param string $newClassName The new class name
+     * @param bool $useBase        Whether to use the base data class for $dataClass
+     */
+    protected function updateLegacyClassNames($dataClass, $fieldName, $oldClassName, $newClassName, $useBase = true)
+    {
+        $schema = DataObject::getSchema();
+        $baseDataClass = $useBase ? $schema->baseDataClass($dataClass) : $dataClass;
+
+        // Check first to ensure that the class has the specified field to update
+        if (!$schema->databaseField($baseDataClass, $fieldName, false)) {
+            return;
+        }
+
+        // Load a list of any records that have obsolete class names
+        $badRecordCount = DataObject::get($baseDataClass)
+            ->filter([$fieldName => $oldClassName])
+            ->count();
+
+        if (!$badRecordCount) {
+            return;
+        }
+
+        if (Director::is_cli()) {
+            echo " * Correcting {$badRecordCount} obsolete {$fieldName} values for {$newClassName}\n";
+        } else {
+            echo "<li>Correcting {$badRecordCount} obsolete {$fieldName} values for {$newClassName}</li>\n";
+        }
+        $table = $useBase ? $schema->baseDataTable($dataClass) : $schema->tableName($dataClass);
```

Comment:
I think `$schema->tableName($baseDataClass)` should always be fine here?

Reply:
Yep you're right!